### PR TITLE
Feat(variables-scss): Add shadow and gradient color variables

### PR DESCRIPTION
**Pull request by pavelklibani**
### Additional context

> [!TIP]
> Can be tested with complex colors:
> Design system name: **Test Complex Shadow Gradients**

### Issue reference

[Exporter | Replace hardcoded color in shadows and gradients with variables](https://jira.almacareer.tech/browse/DS-1473)

### CODE REVIEW

File: exporters/variables-scss/src/helpers/colorHelper.ts [hunk 1]
```diff
@@ -42,3 +43,31 @@ export const normalizeColor = (hexCode: string): string => {
...
+};
+
+// TODO: This is a temporary solution. This function can be removed when Supernova supports variables.
+export const transformColorsToVariables = (name: string, value: string, groupName?: string): string => {
+  let counter = 1;
+  const spiritVariablePrefix = 'spirit-color';
+  const renderGroupName = groupName ? `${groupName}-` : '';
+
+  const transformedValue = value
+    .split(',') // Split into multiple shadows if present
```
Comment (pavelklibani) [inline exporters/variables-scss/src/helpers/colorHelper.ts:59]: i'll fix this comment

File: exporters/variables-scss/src/helpers/colorHelper.ts [hunk 2]
```diff
@@ -42,3 +43,31 @@ export const normalizeColor = (hexCode: string): string => {
 
   return `#${processedHex}`;
 };
+
+// Find all hex colors in string and normalize them
```
Comment (literat) [inline exporters/variables-scss/src/helpers/colorHelper.ts:47]: Are the comments needed? Shouldn't the function name be named `findAllHexColorsInStringAndNormalize`? I am ok with it :-)

Just the comment suggests that the name of the function is not enough to understand what the function does.

File: exporters/variables-scss/src/helpers/colorHelper.ts [hunk 3]
```diff
@@ -42,3 +43,31 @@ export const normalizeColor = (hexCode: string): string => {
...
+export const findAndNormalizeColors = (input: string): string => {
+  return input.replace(HEX_COLOR_REGEX, (match) => normalizeColor(match.slice(1)));
+};
+
+// TODO: This is a temporary solution. This function can be removed when Supernova supports variables.
```
Comment (literat) [inline exporters/variables-scss/src/helpers/colorHelper.ts:52]: Please, use `@TODO` which is compatible with JsDoc style.

File: exporters/variables-scss/src/helpers/colorHelper.ts [hunk 4]
```diff
@@ -42,3 +43,31 @@ export const normalizeColor = (hexCode: string): string => {
...
+export const transformColorsToVariables = (name: string, value: string, groupName?: string): string => {
+  let counter = 1;
+  const spiritVariablePrefix = 'spirit-color';
```
Comment (literat) [inline exporters/variables-scss/src/helpers/colorHelper.ts:55]:
```suggestion
  const spiritVariableColorPrefix = 'spirit-color';
```

**Review by curdaj — APPROVED** [review]

File: exporters/variables-scss/src/helpers/colorHelper.ts (hunk 2 above)
Comment (pavelklibani) [reply exporters/variables-scss/src/helpers/colorHelper.ts:47]: yes, i use alot of comments during the development when i moving things around, we dont need them in the code. i'll remove them

**Review by crishpeen — APPROVED** [review]
LGTM, the output code works. The only problem is with the linter.

Please also do a rebase, so the typograhy quotes and included in this PR and in the exporter.cjs is changed.

**Pull request merged**